internal/utils/concurrency.go

```diff
@@ -0,0 +1,17 @@
+package utils
```

Comment:
**Non-blocking idea:** Would you consider having this concurrency package nested inside `internal` instead of being in the generic `utils` directory? It might make for a nicer import and implementation:

```
pool := concurrency.NewPool(ctx, int(l.resolveNodeBreadthLimit))
```

Or possibly:

```
concurrencyPool := pool.New(ctx, int(l.resolveNodeBreadthLimit))
```

Reply:
Moved it to `internal/concurrency`